vision_ocv/benchmark/modules/src/BM_ocv_vision_factory_test.cpp

```diff
@@ -0,0 +1,136 @@
+// Copyright 2023 Klepsydra Technologies AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <benchmark/benchmark.h>
+#include <spdlog/spdlog.h>
+#include <opencv2/opencv.hpp>
+#include <image_event_stamped.h>
+#include <klepsydra/mem_core/basic_middleware_provider.h>
+#include <klepsydra/vision_ocv/image_data_factory.h>
+#include <klepsydra/vision_ocv/file_image_stream_service.h>
+#include <klepsydra/core/event_transform_forwarder.h>
+
+class FactoryPerformanceTest : public benchmark::Fixture {
+    public:
+        FactoryPerformanceTest() :
+         benchmark::Fixture(),
+         width(1000), 
+         height(500),
+         imageDataFactory(width, height,  cv::IMREAD_REDUCED_GRAYSCALE_2, "raw_image"){
+
+        }
+        ~FactoryPerformanceTest() {}
+        
+        void SetUp(const ::benchmark::State& state) {
+            bool withFactory = state.range(0);
+            if (withFactory) {
+                mp.reset(new kpsr::mem::BasicMiddlewareProvider<kpsr::vision_ocv::ImageData>(nullptr,
+                "img", 4, 6, imageDataFactory.initializerFunction, imageDataFactory.eventClonerFunction, false));
+            } else {
+                mp.reset(new kpsr::mem::BasicMiddlewareProvider<kpsr::vision_ocv::ImageData>(nullptr,
+                "img", 4, 6, nullptr, nullptr, false));
```

Comment:
Let's do as @mharshe suggested:
`poolSize = 0`, and clone `std::function `to call the` cv::Mat::copy`